samples/grids/grid/binding-remote-data/App.razor

```diff
@@ -0,0 +1,158 @@
+                       DataType="GridColumnDataType.Number"
+                       Editable="true"
+                       Resizable="true">
+            </IgbColumn>
+
+            <IgbColumn Field="UnitsInStock"
+                       Header="Units in Stock"
+                       DataType="GridColumnDataType.Number"
+                       Editable="true"
+                       Resizable="true">
+            </IgbColumn>
+
+            <IgbColumn Field="OrderDate"
+                       Header="Order Date"
+                       DataType="GridColumnDataType.Date"
+                       Editable="true"
+                       Resizable="true">
+            </IgbColumn>
+
+            <IgbColumn Field="Discontinued"
+                       Header="Discontinued"
+                       DataType="GridColumnDataType.Boolean"
+                       Editable="true">
+            </IgbColumn>
+
+            <IgbColumn Field="ReorderLevel"
+                       Header="Reorder Level"
+                       DataType="GridColumnDataType.Number"
+                       Editable="true"
+                       Filterable="true">
+            </IgbColumn>
+        </IgbGrid>
+
+    </div>
+</div>
+
+@code {
+    protected List<NwindDataItem> dataSource1Table1 { get; set; }
+    protected List<NwindDataItem[]> partitionedProducts { get; set; }
+    private IgbGrid grid1;
+    Random random = new Random();
+    protected int Page { get; set; } = 0;
+    protected int PerPage { get; set; } = 5;
+    protected int TotalRecords { get; set; } = 77;
+    protected bool loading = false;
+
+    protected override async Task OnInitializedAsync()
+    {
+        dataSource1Table1 = await dataSource1Service.GetTable1();
+        partitionedProducts = (from product in dataSource1Table1 select product)
+                                 .Chunk(PerPage)
+                                 .ToList();
+        dataSource1Table1 = partitionedProducts.FirstOrDefault().ToList();
+    }
+    protected override async Task OnAfterRenderAsync(bool firstRender)
+    {
+        var grid1 = this.grid1;
+        await grid1.EnsureReady();
+    }
+
+    public async void OnPagingDone(IgbPageEventArgs e)
+    {
+        Console.WriteLine($"Page: {e.Detail.Current}");
+        this.Page = (int)e.Detail.Current;
+        this.loading = true;
+        await PopulateData();
+    }
+
+    public async Task PopulateData()
+    {
+        try
+        {
+            await foreach (var item in FetchItems())
+            {
+                dataSource1Table1 = partitionedProducts[Page].ToList();
+                StateHasChanged();
+            }
+            loading = false;
+
+        }
+        catch (Exception _ex)
+        {
+
+        }
+    }
+
+    public async IAsyncEnumerable<int> FetchItems()
+    {
+        for (int i = 1; i <= PerPage; i++)
+        {
+            yield return (Page) * PerPage + i;
+        }
+    }
+
+    public class DataSource1Service : IDataSource1Service
```

Comment:
rename class to ProductDataService and move class to separate file:
.\Services\ProductDataService.cs